doc/compiler/debug/DebugCounters.md

```diff
@@ -0,0 +1,156 @@
+
+### Colon
+
+`:`
+
+A colon in a counter name is similar to a slash, except it indicates that the denominator should not be computed automatically. Rather, the developer will insert the denominator counter separately. The colon simply indicates that the user wants to compute the ratio between two counters. Asking for a counter with a name of the form "a:b" actually provides two counters: "a" and "a:b". The ratio between these two counters will be displayed in the counter report, but increments to "a:b" will not be counted as increments to "a".
+
+For example:
+```
+prependDebugCounter("frames", tt);
+...
+prependDebugCounter("frames:#autos", tt, numAutos);
+```
+This produces two counters that are unrelated except that the jit will report their ratio when the run is complete:
+```
+1: frames         |  3914036134 |   __   1 __
+2: frames:#autos  | 22680880758 |      5.79  |  __   2 __
+```
+Looking down from the "__ 1__" heading, you can see that each frame has an average of 5.79 autos.
+
+### Equals
+
+`=`
+
+The equals sign indicates to the debug counter display routine that the remainder of the counter name should be **sorted numerically** rather than alphabetically. It has no other special meaning.
+
+### Parentheses
+
+`()`
+
+Parentheses are used to mark a **verbatim section** of a counter name, within which the other special characters lose their special meaning. For example, asking for a counter with a name of the form "(a/b)" produces just one counter. Without the special meaning of parentheses, this would have produced two counters: "(a" and "(a/b)".
+
+Parentheses can be nested.
+
+### Hash
+
+`#`
+
+The hash symbol has no particular special meaning, but is used by convention to signify counters that are **suitable for producing a histogram**. Generally you can include a hash symbol in any debug counter that is incremented by some value other than 1, such as "#parameters".
+
+### Other naming considerations
+
+Counters can have arbitrary names, though care should be taken with certain special characters:
+- Whitespace and other shell-control characters may make them awkward to specify on the command line
+- Characters with special meanings in the JIT Compiler's regular expressions will make them awkward to filter
+
+## Specifying Output
+
+### Environment Variables
+
+```
+TR_DebugCounterFileName
+````
+Specifies the file to which the debug counter output should be appended (creating the file if necessary), relative to the current working directory. If not specified, debug counter output is written to stdout.
+
+### JIT Options
+
+A full listing of debug counter-related JIT options can be found by using a command like the following:
+```
+java -Xjit:help={*debugCounter*} -version
```

Comment:
Change this to be more in line with how one would specify options in OMR.

Reply:
86cd7418be129d45c6e4a8eeaf4c4f9677a459f2